internal/config/config.go

```diff
@@ -393,6 +418,57 @@ func addressPoolFromCR(p metallbv1beta1.IPPool, bgpCommunities map[string]uint32
 	return ret, nil
 }
 
+func addressPoolFromLegacyCR(p metallbv1beta1.AddressPool, bgpCommunities map[string]uint32) (*Pool, error) {
+	if p.Name == "" {
+		return nil, errors.New("missing pool name")
+	}
+
+	ret := &Pool{
+		AvoidBuggyIPs: p.Spec.AvoidBuggyIPs,
+		AutoAssign:    true,
+	}
+
+	if p.Spec.AutoAssign != nil {
+		ret.AutoAssign = *p.Spec.AutoAssign
+	}
+
+	if len(p.Spec.Addresses) == 0 {
+		return nil, errors.New("pool has no prefixes defined")
+	}
+
+	cidrsPerAddresses := map[string][]*net.IPNet{}
+	for _, cidr := range p.Spec.Addresses {
+		nets, err := ParseCIDR(cidr)
+		if err != nil {
+			return nil, fmt.Errorf("invalid CIDR %q in pool %q: %s", cidr, p.Name, err)
+		}
+		ret.CIDR = append(ret.CIDR, nets...)
+		cidrsPerAddresses[cidr] = nets
+	}
+	switch Proto(p.Spec.Protocol) {
+	case Layer2:
+		if len(p.Spec.BGPAdvertisements) > 0 {
+			return nil, errors.New("cannot have bgp-advertisements configuration element in a layer2 address pool")
+		}
+		ret.L2Advertisements = []*L2Advertisement{{}}
+	case BGP:
+		ads, err := bgpAdvertisementsFromLegacyCR(p.Spec.BGPAdvertisements, cidrsPerAddresses, bgpCommunities)
+		if err != nil {
+			return nil, fmt.Errorf("parsing BGP communities: %s", err)
```

Comment:
The err msg here is misleading. bgpAdvertisementsFromLegacyCR func parsing not just communities.

Reply:
Changed